### PR TITLE
more datatype→schemaName renaming

### PR DESCRIPTION
**Pull request by jtbandes**
**User-Facing Changes**
None

**Description**
Rename datatype in AdvertiseOptions and some local variables

### CODE REVIEW

File: benchmark/src/players/SinewavePlayer.ts [hunk 1]
```diff
@@ -95,7 +95,7 @@ class SinewavePlayer implements Player {
 
     for (let i = 0; i < sinewaveCount; ++i) {
       const topicName = `sinewave_${i}`;
-      topics.push({ name: topicName, datatype: "Sinewave" });
```
Comment (defunctzombie) [inline benchmark/src/players/SinewavePlayer.ts:98]: How did these get past the type change? Do we have something downstream that ensures we continue setting `datatype` even when a player only sets `schemaName` ?

**Review by defunctzombie — APPROVED** [review]
As long as we have something downstream that sets `datatype` for extension panels 👍

File: benchmark/src/players/SinewavePlayer.ts (hunk 1 above)
Comment (jtbandes) [reply benchmark/src/players/SinewavePlayer.ts:98]: I don't think the benchmark players are being typechecked in CI.

**Comment by jtbandes** [comment]
https://github.com/foxglove/studio/blob/9e768524cb65b393087ffbeb5f88444b9eb06e8a/packages/studio-base/src/components/PanelExtensionAdapter/renderState.ts#L118-L122

**Comment by jtbandes** [comment]
As discussed in the https://github.com/foxglove/studio/pull/4582 description.

> Within Studio code, we use a Topic type that has only name and schemaName. The deprecated datatype key is added by the ExtensionAdapter for extension panels.

**Pull request merged**